### PR TITLE
Feature/issue 578 documentation for diagnose utility

### PR DESCRIPTION
**Pull request by mitzimorris**
#### Submisison Checklist

- [x] Run tests: `./runCmdStanTests.py src/test`
- [x] Declare copyright holder and open-source license: see below

#### Summary:
add chapter on CmdStan's `diagnose` tool to manual.

#### Intended Effect:
better documentation

#### How to Verify:
`make manual`

#### Side Effects:
n/a

#### Documentation:
self-documenting.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):  Columbia University

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

**Comment by mitzimorris** [comment]
as I created the PR, copyright goes to Columbia.
how should we credit/thank James Ramsey?

**Comment by bob-carpenter** [comment]
Up to you.

File: src/docs/cmdstan-guide/diagnose.tex [hunk 1]
```diff
@@ -0,0 +1,112 @@
+\chapter{{\tt\bfseries diagnose}: Output Sanity Checking}\label{diagnose.chapter}
```
Comment (betanalpha) [inline src/docs/cmdstan-guide/diagnose.tex]: Please rename the section to "Diagnosing Biased Hamiltonian Monte Carlo Inferences"

File: src/docs/cmdstan-guide/diagnose.tex [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+\begin{itemize}
+\item Transitions that hit the maximum treedepth
+\item Divergent transitions
+\item Low E-BFMI values
+\item Low effective sample sizes
+\item High $\hat{R}$ values
+\end{itemize}
+
+The meanings of several of these problems are discussed in
+\url{http://mc-stan.org/misc/warnings.html#runtime-warnings}.
```
Comment (betanalpha) [inline src/docs/cmdstan-guide/diagnose.tex]: Additionally please include a reference to the review paper, https://arxiv.org/abs/1701.02434.

File: src/docs/cmdstan-guide/diagnose.tex [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+% This section cribbed from a similar section in stansummary
```
Comment (betanalpha) [inline src/docs/cmdstan-guide/diagnose.tex]: This and the other LaTeX comments are not necessary.

File: src/docs/cmdstan-guide/diagnose.tex [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+  The E-BFMI, 0.27, is below the nominal threshold of 0.3 which
+  suggests that HMC may have trouble exploring the target
+  distribution.  You should consider any reparameterizations if
+  possible.
+\end{Verbatim}
+\caption{Example output from \code{bin/diagnose}.}
+\label{bin-diagnose-eg.figure}
+\end{figure}
+
```
Comment (betanalpha) [inline src/docs/cmdstan-guide/diagnose.tex]: Please add a remark that resolutions of the other diagnostic warnings are suggested in the above references but that in general resolutions will be model specific and hence general guidelines are impossible.

**Comment by mitzimorris** [comment]
Jenkins, retest this please.

**Comment by mitzimorris** [comment]
made changes requested per code review.  merging.

**Pull request merged**